receiver/otlpreceiver/internal/trace/otlp.go

```diff
@@ -49,7 +51,30 @@ func (r *Receiver) Export(ctx context.Context, req ptraceotlp.ExportRequest) (pt
 	// NonPermanent errors will be converted to codes.Unavailable (equivalent to HTTP 503)
 	// Permanent errors will be converted to codes.InvalidArgument (equivalent to HTTP 400)
 	if err != nil {
-		return ptraceotlp.NewExportResponse(), errors.GetStatusFromError(err)
+		// See if an error is retryable.
+		cet := consumererror.Traces{}
+		if errors.As(err, &cet) {
+			retries := 3
+
+			// Retry three times; the real logic would be more complex akin to consumerretry.
+			for i := 0; i < retries; i++ {
+				// Re-use the data that was given to be retried.
+				td = cet.Data()
+
+				// Update the context to include a key that indicates the component to be retried.
+				retryCtx := context.WithValue(ctx, consumererror.ComponentIDKey, cet.ID())
```

Comment:
I feel like the only reason we would want to use context here is because we don't want to change the existing APIs. I think an options pattern would make more sense going forward to apply component retry filters like this.